src/protocols/echo/EchoClient.cpp

```diff
@@ -0,0 +1,142 @@
+/*
+ *
+ *    Copyright (c) 2020 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements an object for a CHIP Echo unsolicitied
+ *      initiator (client).
+ *
+ */
+
+#include "Echo.h"
+
+namespace chip {
+namespace Protocols {
+
+EchoClient::EchoClient()
+{
+    ExchangeMgr            = nullptr;
+    OnEchoResponseReceived = nullptr;
+    ExchangeCtx            = nullptr;
+}
+
+CHIP_ERROR EchoClient::Init(ExchangeManager * exchangeMgr)
+{
+    // Error if already initialized.
+    if (ExchangeMgr != nullptr)
+        return CHIP_ERROR_INCORRECT_STATE;
+
+    ExchangeMgr            = exchangeMgr;
+    OnEchoResponseReceived = nullptr;
```

Comment:
these are set to null in the constructor. Is the extra null setting needed?

Reply:
This allow the application re-init the EchoClient without always creating an new one.